app/assets/javascripts/ramp_utils.js

```diff
@@ -360,3 +360,9 @@ function closeAlert() {
     $('#playlistitem_scope_track').prop('checked', true);
   }
 }
+
+/** Refresh stream token by reloading active m3u8 */
+function m3u8Reload() {
+  player = document.getElementById('iiif-media-player');
+  player.player.src(player.player.currentSources());
+};
```

Comment:
For this scenario (user is still online), I think we could avoid resetting the player and instead just fetch the auto.m3u8:
```
fetch(player.player.currentSources()[0]["src"]);
```
For the other scenario (token already expired) I think we would need to set the player sources like you're doing.

Reply:
In testing the fetch this morning, it looks like it does actually work for both cases. That would probably be better than fully resetting the player and looks like it removes the need for the `on(loadeddata, ...)` listener.